ember-server/src/test/scala/org/http4s/ember/server/EmberServerSuite.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.server
+
+import cats.syntax.all._
+import cats.effect._
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.server.Server
+import org.http4s.implicits._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.ember.client.EmberClientBuilder
+
+import scala.concurrent.duration._
+
+class EmberServerSuite extends Http4sSuite {
+
+  def service[F[_]](implicit F: Async[F]): HttpApp[F] = {
+    val dsl = new Http4sDsl[F] {}
+    import dsl._
+
+    import scala.concurrent.duration._
+
+    HttpRoutes
+      .of[F] { case GET -> Root =>
+        Ok("Hello!")
+      }
+      .orNotFound
+  }
+
+  def serverResource: Resource[IO, Server[IO]] =
+    EmberServerBuilder
+      .default[IO]
+      .withHttpApp(service[IO])
+      .build
+
+  def client: FunFixture[Client[IO]] =
+    ResourceFixture(EmberClientBuilder.default[IO].build)
+
+  def server: FunFixture[Server[IO]] =
+    ResourceFixture(serverResource)
+
+  def fixture: FunFixture[(Server[IO], Client[IO])] =
+    FunFixture.map2(server, client)
+
+  fixture.test("server shutdown waits for outstanding responses to complete") {
+    case (server, client) =>
+      IO.sleep(3.seconds) >> client
```

Comment:
For some reason the server isn't immediately connect-able to after the resource is acquired. I wonder if there's some detail in the NIO API about this, but I _have_ to wait 3 seconds or else I get a `Connection refused`, which seems to imply that the server socket channel hasn't even been opened yet?